src/handson/Table.js

```diff
@@ -0,0 +1,351 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "handsontable", "../common/Widget", "../common/HTMLWidget", "../common/PropertyExt", "../common/Utility", "css!handsontable", 
+
+"css!./Table"], factory);
+    } else {
+        root.handson_Table = factory(root.d3, root.Handsontable, root.common_Widget, root.common_HTMLWidget, root.common_PropertyExt, root.common_Utility);
+    }
+}(this, function (d3, Handsontable, Widget, HTMLWidget, PropertyExt, Utility) {
+
+    function Column(owner) {
+        PropertyExt.call(this);
+        this._owner = owner;
+    }
+    Column.prototype = Object.create(PropertyExt.prototype);
+    Column.prototype.constructor = Column;
+    Column.prototype._class += " handson_Table.Column";
+
+    Column.prototype.publish("tableColumns", "", "set", "Only show column properties of the table",function () { return this._owner ? this._owner.columns() : []; }, { optional: true });
+    Column.prototype.publish("minRange", null, "number", "Minimum value of a handsontable", function () { return this._owner ? this._owner.columns() : []; }, { tags: ["Basic"], optional: true });
+    Column.prototype.publish("maxRange", null, "number", "Maximum value of a handsontable", function () { return this._owner ? this._owner.columns() : []; }, { tags: ["Basic"], optional: true });
+    Column.prototype.publish("belowMinRangeColor", "#ff0000", "html-color", "Series Color", function () { return this._owner ? this._owner.columns() : []; }, { optional: true });
+    Column.prototype.publish("aboveMaxRangeColor", "#0000ff", "html-color", "Series Color", function () { return this._owner ? this._owner.columns() : []; }, { optional: true });
+    Column.prototype.publish("rangeColor", "#00ff00", "html-color", "Series Color", function () { return this._owner ? this._owner.columns() : []; }, { optional: true });
+    
+    function Table() {
+        HTMLWidget.call(this);
+        this._tag = "div";
+
+        this._selectionBag = new Utility.Selection();
+        this._widgetCache = {};
+        this._widgetCache2 = {};
+    }
+//handson
+    Table.prototype = Object.create(HTMLWidget.prototype);
+    Table.prototype.constructor = Table;
+    Table.prototype._class += " handson_Table";
+    Table.prototype.Column = Column;
+    Table.prototype.publish("mappings", [], "propertyArray", "Source Columns", null, { autoExpand: Column });
+    Table.prototype.publish("renderHtmlDataCells", false, "boolean", "enable or disable HTML within cells",null,{tags:["Private"]});
+    Table.prototype.publish("fixedColumn", false, "boolean", "Enable or disable fixed first column", null, { tags: ["Private"] });
```

Comment:
I have changed it from private to basic. Please let me know what is the exact difference between them here.